Categories/Power.lua

```diff
@@ -25,24 +29,64 @@ if Enum and Enum.PowerType then
 	Enum.PowerType.Mana, Enum.PowerType.Runes, Enum.PowerType.Chi, Enum.PowerType.LunarPower, Enum.PowerType.SoulShards, Enum.PowerType.ArcaneCharges
 	SPELL_POWER_BURNING_EMBERS, SPELL_POWER_HOLY_POWER, SPELL_POWER_LIGHT_FORCE, SPELL_POWER_SHADOW_ORBS =
 	Enum.PowerType.Obsolete, Enum.PowerType.HolyPower, Enum.PowerType.Obsolete, Enum.PowerType.Obsolete
+	SPELL_POWER_RAGE, SPELL_POWER_FOCUS, SPELL_POWER_ENERGY, SPELL_POWER_COMBO_POINTS, SPELL_POWER_RUNIC_POWER, SPELL_POWER_LUNAR_POWER =
+    Enum.PowerType.Rage, Enum.PowerType.Focus, Enum.PowerType.Energy, Enum.PowerType.ComboPoints, Enum.PowerType.RunicPower, Enum.PowerType.LunarPower
+	SPELL_POWER_MAELSTROM, SPELL_POWER_INSANITY, SPELL_POWER_FURY, SPELL_POWER_PAIN =
+    Enum.PowerType.Maelstrom, Enum.PowerType.Insanity, Enum.PowerType.Fury, Enum.PowerType.Pain
 end
 
+-- copied from Blizzard_CombatText, which isn't exposed anywhere else
+local powerEnumFromStringLookup =
+{
+	[MANA:lower()] = SPELL_POWER_MANA,
+	[RAGE:lower()] = SPELL_POWER_RAGE,
+	[FOCUS:lower()] = SPELL_POWER_FOCUS,
+	[ENERGY:lower()] = SPELL_POWER_ENERGY,
+	[COMBO_POINTS:lower()] = SPELL_POWER_COMBO_POINTS,
+	[RUNES:lower()] = SPELL_POWER_RUNES,
+	[RUNIC_POWER:lower()] = SPELL_POWER_RUNIC_POWER,
+	[SOUL_SHARDS:lower()] = SPELL_POWER_SOUL_SHARDS,
+	[LUNAR_POWER:lower()] = SPELL_POWER_LUNAR_POWER,
+	[HOLY_POWER:lower()] = SPELL_POWER_HOLY_POWER,
+	-- [ALTERNATE:lower()] = SPELL_POWER_ALTERNATE,
+	[MAELSTROM:lower()] = SPELL_POWER_MAELSTROM,
+	[CHI:lower()] = SPELL_POWER_CHI,
+	[INSANITY:lower()] = SPELL_POWER_INSANITY,
+	[ARCANE_CHARGES:lower()] = SPELL_POWER_ARCANE_CHARGES,
+	[FURY:lower()] = SPELL_POWER_FURY,
+	[PAIN:lower()] = SPELL_POWER_PAIN,
+}
+
 DogTag:AddTag("Unit", "MP", {
-	code = UnitPower,
+	code = function(unit, type)
+		local enum
+		if type then
+			enum = powerEnumFromStringLookup[type:lower()]
+		end
+		return UnitPower(unit, enum)
+	end,
 	arg = {
-		'unit', 'string;undef', 'player'
+		'unit', 'string;undef', 'player',
+		'type', 'string;undef', '@undef',
 	},
 	ret = "number",
 	events = "UNIT_POWER_FREQUENT#$unit;UNIT_DISPLAYPOWER#$unit",
 	doc = L["Return the current mana/rage/energy of unit"],
-	example = ('[MP] => "%d"'):format(UnitPowerMax("player")*.632),
+	example = ('[MP] => "%d"'):format(UnitPower("player")*.632),
```

Comment:
Might want to add an example showing usage of the type parameter (and likewise for the other tags where the parameter was added)